packages/shared/src/hooks/post/usePostShareLoop.ts

```diff
@@ -0,0 +1,42 @@
+import { useState } from 'react';
+import { useActiveFeedNameContext } from '../../contexts';
+import { useFeature } from '../../components/GrowthBookProvider';
+import { feature } from '../../lib/featureManagement';
+import { useMutationSubscription } from '../mutationSubscription';
+import { upvoteMutationKey, UseVotePostMutationProps } from '../vote';
+import { Post } from '../../graphql/posts';
+
+interface UsePostShareLoop {
+  shouldShowOverlay: boolean;
+  onInteract: () => void;
+}
+
+export const usePostShareLoop = (post: Post): UsePostShareLoop => {
+  const { feedName } = useActiveFeedNameContext();
+  const [justUpvoted, setJustUpvoted] = useState(false);
+  const [hasInteracted, setHasInteracted] = useState(false);
+  const shareLoopsEnabled = useFeature(feature.shareLoops);
+  const shouldShowOverlay = justUpvoted && !hasInteracted && shareLoopsEnabled;
+
+  useMutationSubscription({
+    matcher: ({ status, mutation }) => {
+      const key = [...upvoteMutationKey, { feedName }];
+
+      return (
+        status === 'success' &&
+        mutation?.options?.mutationKey?.toString() === key.toString()
```

Comment:
`key` is not used anywhere else and here we conver it with `toString()`. Seems like it would be more efficient to assign `key` with the `.toString()` already executed. That way we only do the conversion once on the assignment.

Reply:
I think this is probably to keep it easier to understand that both of them should be converted and not be missed. Wdyt? Should we add the `toString` on the memoized variable?